include/drivers/hwinfo.h

```diff
@@ -54,6 +70,30 @@ __syscall ssize_t hwinfo_get_device_id(u8_t *buffer, size_t length);
 
 ssize_t z_impl_hwinfo_get_device_id(u8_t *buffer, size_t length);
 
+/**
+ * @brief      Retrieve cause of device reset.
+ *
+ * @param      cause  OR'd flags TBD
+ *
+ * @retval zero if successful.
+ * @retval -ENOTSUP if there is no implementation for the particular device.
+ * @retval any negative value on driver specific errors.
+ */
+__syscall int hwinfo_get_reset_cause(u32_t *cause);
+
+int z_impl_hwinfo_get_reset_cause(u32_t *cause);
+
+/**
+ * @brief      Clear cause of device reset.
+ *
+ * @retval zero if successful.
```

Comment:
This might not be the right API for clearing.  In my applications I need the ability to clear certain reasons, leaving the others in-place for the next reset.

Reply:
That depends on the SoC. Some will clear the entire reason register as soon as it is read. Others let you clear specific bits.